trapper/commands.py

```diff
@@ -148,7 +241,12 @@ def add_subcommands():
         add_subcommands()
 
     # Now we can parse the arguments.
-    args = parser.parse_args()
+    if argv[0] == "distributed-run":
+        dist_args = torch_distributed_parse_args()
+    else:
+        dist_args = argparse.Namespace()
+    args, _ = parser.parse_known_args()
+    args = merge_args_safe(args, dist_args)
```

Comment:
Can't we simplify the logic here? E.g., make the Line 248 as the first statement here. Then, add the conditional in 244. In the if True branch, parse the distributed args and update the `args` variable. Delete the else branch.